### PR TITLE
Adding auth singleton references to abstract models #17

### PR DESCRIPTION
**Pull request by kawika-brennan-bfg**
For each of the abstract DataFrame models, there is now a reference to the Authenticator Singleton as 'authenticator'.

### CODE REVIEW

File: gamebench_api_client/models/dataframes/generic/abstract_generic.py [hunk 1]
```diff
@@ -21,6 +22,7 @@ def __init__(self, **request_parameters):
                 build and send a response.
         """
 
+        self.authenticator = Authenticator(**request_parameters)
```
Comment (aaron-wilson-bfg) [inline gamebench_api_client/models/dataframes/generic/abstract_generic.py]: Let's fetch the parameters for Authenticator's init from settings.

Comment (kawika-brennan-bfg) [reply gamebench_api_client/models/dataframes/generic/abstract_generic.py]: Thank you, I'll change that.

File: gamebench_api_client/global_settings.py [hunk 2]
```diff
@@ -3,8 +3,8 @@
 GAMEBENCH_CONFIG = {
     'url': 'https://api.production.gamebench.net',
     'api_version': '/v1',
-    'username': '',
-    'password': '',
+    'username': 'john.smith',
```
Comment (aaron-wilson-bfg) [inline gamebench_api_client/global_settings.py]: Are these placeholder values necessary in the production code?

Comment (kawika-brennan-bfg) [reply gamebench_api_client/global_settings.py]: No, I just thought placeholder might help the visual. I will remove it to avoid any later confusion!

**Comment by aaron-wilson-bfg** [comment]
Hi @kawika-brennan-bfg 
Is this PR ready for review again?

**Comment by kawika-brennan-bfg** [comment]
> Hi @kawika-brennan-bfg
> Is this PR ready for review again?

Yes!

File: gamebench_api_client/models/dataframes/generic/abstract_generic.py (hunk 1 above)
Comment (aaron-wilson-bfg) [reply gamebench_api_client/models/dataframes/generic/abstract_generic.py]: 👍

File: gamebench_api_client/global_settings.py (hunk 2 above)
Comment (aaron-wilson-bfg) [reply gamebench_api_client/global_settings.py]: I'm ready to approve once this change is pushed.

Comment (kawika-brennan-bfg) [reply gamebench_api_client/global_settings.py]: Removed in 0a40931

Comment (aaron-wilson-bfg) [reply gamebench_api_client/global_settings.py]: Thanks! I wonder why it isn't showing in the "view changes" view? I went to the file directly to confirm.

**Pull request merged**